File: mobile/android/app/src/main/java/com/volttracker/obdpoc/AppStateJson.java
package com.volttracker.obdpoc;

import org.json.JSONObject;

/**
 * Assembles the {@code setAppState} JSON payload pushed to the dashboard from the latest telemetry,
 * status, and storage snapshots. Extracted from {@link MainActivity} as a pure function so the
 * payload shape stays unit-testable.
 */
final class AppStateJson {

    private AppStateJson() {}

    static String build(
            String version,
            boolean bluetoothReady,
            boolean locationGranted,
            boolean notificationsGranted,
            String lastAddress,
            String lastName,
            JSONObject telemetry,
            JSONObject status,
            JSONObject storage) {
        return new AppStatePayload(
                        version,
                        bluetoothReady,
                        locationGranted,
                        notificationsGranted,
                        lastAddress,
                        lastName,
                        telemetry,
                        status,
                        storage)
                .toJson()
                .toString();
    }
}

File: mobile/android/app/src/main/kotlin/com/volttracker/obdpoc/AppStateJson.kt
package com.volttracker.obdpoc

import org.json.JSONObject

/**
 * Assembles the `setAppState` JSON payload pushed to the dashboard from the latest telemetry,
 * status, and storage snapshots. Extracted from [MainActivity] as a pure function so the payload
 * shape stays unit-testable.
 */
object AppStateJson {
    @JvmStatic
    fun build(
        version: String?,
        bluetoothReady: Boolean,
        bluetoothPermissionGranted: Boolean,
        bluetoothEnabled: Boolean,
        locationGranted: Boolean,
        notificationsGranted: Boolean,
        lastAddress: String?,
        lastName: String?,
        telemetry: JSONObject?,
        status: JSONObject?,
        storage: JSONObject?,
    ): String =
        AppStatePayload(
            version,
            bluetoothReady,
            bluetoothPermissionGranted,
            bluetoothEnabled,
            locationGranted,
            notificationsGranted,
            lastAddress,
            lastName,
            telemetry,
            status,
            storage,
        ).toJson().toString()
}
